Source/WebCore/loader/NavigationAction.cpp

```diff
@@ -131,6 +136,21 @@ NavigationAction::NavigationAction(Document& requester, const ResourceRequest& o
     setIsRequestFromClientOrUserInput(isRequestFromClientOrUserInput);
 }
 
+NavigationAction::NavigationAction(const NavigationRequester& requester, const ResourceRequest& originalRequest, InitiatedByMainFrame initiatedByMainFrame, bool isRequestFromClientOrUserInput, FrameLoadType frameLoadType, bool isFormSubmission, Event* event, ShouldOpenExternalURLsPolicy shouldOpenExternalURLsPolicy, const AtomString& downloadAttribute, Element* sourceElement)
```

Comment:
Can we have the constructor above call your new constructor to avoid duplication?
e.g.
```
NavigationAction::NavigationAction(Document& requester, const ResourceRequest& originalRequest, InitiatedByMainFrame initiatedByMainFrame, bool isRequestFromClientOrUserInput, FrameLoadType frameLoadType, bool isFormSubmission, Event* event, ShouldOpenExternalURLsPolicy shouldOpenExternalURLsPolicy, const AtomString& downloadAttribute, Element* sourceElement)
    : NavigationAction(NavigationRequester::from(requester), originalRequest, /* ... */)
{ }
```